app/controllers/api/account.php

```diff
@@ -201,19 +201,48 @@ function sendSessionAlert(Locale $locale, Document $user, Document $project, Doc
         default => throw new Exception(Exception::USER_INVALID_TOKEN)
     });
 
+    $sessionData = [
+        '$id' => ID::unique(),
+        'userId' => $user->getId(),
+        'userInternalId' => $user->getSequence(),
+        'provider' => Auth::getSessionProviderByTokenType($verifiedToken->getAttribute('type')),
+        'secret' => Auth::hash($sessionSecret), // One way hash encryption to protect DB leak
+        'userAgent' => $request->getUserAgent('UNKNOWN'),
+        'ip' => $request->getIP(),
+        'factors' => [$factor],
+        'countryCode' => ($record) ? \strtolower($record['country']['iso_code']) : '--',
+        'expire' => DateTime::addSeconds(new \DateTime(), $duration)
+    ];
+
+    // For OAuth2 tokens, retrieve and add provider tokens from identity
+    if ($verifiedToken->getAttribute('type') === Auth::TOKEN_TYPE_OAUTH2) {
+        // Find the most recently updated identity for this user with an access token
+        // This will be the identity that was just used in the OAuth2 flow
+        $identities = $dbForProject->find('identities', [
+            Query::equal('userInternalId', [$user->getSequence()]),
+            Query::orderDesc('$updatedAt'),
+            Query::limit(100)
+        ]);
+
+        // Find the first identity with a non-empty access token
+        $latestIdentity = null;
+        foreach ($identities as $identity) {
```

Comment:
If we only need the latest identity, it will be much more efficient to actually query only the last row from the DB. That said, not sure if this is the most solid solution, don't we have a link between session X and identity Y?